src/Text/Layout/Table.hs

```diff
@@ -288,8 +288,16 @@ tableLinesB specs TableStyle { .. } rowHeader colHeader rowGroups =
 
     -- | Intersperse a row with its separators
     withRowSeparators sep r = map (first sep) . flattenHeader . fmap fst $ zipHeader r rowHeaderShape
-    withColSeparators sep r = map (first $ \x -> if isColumnDrawn x then sep x else "") . flattenHeader . fmap fst $ zipHeader r colHeaderShape
-    isColumnDrawn x = not $ null (headerC $ fst x) && null (groupC $ snd x)
+    withColSeparators sep r = map (first drawSep) . flattenHeader . fmap fst $ zipHeader r colHeaderShape
```

Comment:
* Please document.
* `sep` seems to be a function but the parameter does not reflect the fact.
* I don't mind an explanation for different cases. (I.e. single line with comment above.)